### ISSUE TITLE
Unable to signup

### ISSUE THREAD

**Issue by FireAstrox** [yeSpud/CS-372-Project#42]
Upon trying to sign up with a valid username and password this error is given at the top of the webpage:

Error: Invalid `prisma.user.findUnique()` invocation in C:\Users\Rylan\Documents\372\CS-372-Project\backend\api\src\routes\authentication.js:97:31 94 usernameCheck(request.body.username) 95 passwordCheck(request.body.password) 96 → 97 if (await prisma.user.findUnique( error: Environment variable not found: DATABASE_URL. --> schema.prisma:13 | 12 | provider = "mongodb" 13 | url = env("DATABASE_URL") | Validation Error Count: 1

I can provide the server log as well if needed

**Comment by yeSpud** [comment]
Did you create the `.env` file writh the following text as per the read me?

https://github.com/yeSpud/CS-372-Project/blob/main/backend/database/example.env

**Comment by FireAstrox** [comment]
Yes, I copy pasted the `DATABASE_URL="mongodb://root:prisma@localhost:27017/prisma-mongo?authSource=admin&retryWrites=true&w=majority"` from the example.env file into a new file called database.env in the same folder, does the new file need to be named specifically.

I have also tried this with juts `DATABASE_URL="mongodb://localhost:27017"`

**Comment by yeSpud** [comment]
It should be `.env` as per the readme - not `database.env`

**Comment by FireAstrox** [comment]
I have changed the file name to ".env", the documentation is not very clear on this as it only states a new .env file needs to be made not that it also needs to be named just ".env".

After changing and restarting the frontend and the backend the same error still appears

**Comment by yeSpud** [comment]
the intent of stating "create a new `.env` file in `backend/database`..." is for a new file called `.env` to be created in the `backend/database` directory.

As for the same prisma error did you run `pnpm prisma generate` in the `backend/database` directory before restarting the api?

**Comment by FireAstrox** [comment]
Yes, I have done this multiple times all with the same result.

Although that might be the intent with your instructions it is not clear or understandable as someone with little to no experience with this

**Comment by yeSpud** [comment]
Interesting. Ill take a look at your setup come Monday in class

**Comment by yeSpud** [comment]
Issue has adapted to a MongoDB SCRAM error on Windows

**Comment by FireAstrox** [comment]
I have reinstalled MongoDB and still am receiving the same error

**Comment by yeSpud** [comment]
Im currently unsure of a resolution (aside from changing the database to NOT require authentication for connecting).

And to be fair you're not the only one to have this issue on Windows. @Alaskanyoda encountered this issue too, and it was only resolved by switching to Linux.

With that said Id rather figure out how to get it to work on Windows before I mandate that the backend be run on Linux only...

**Comment by Maddie571** [comment]
To clarify confusion with the discord and issue #47, im having the same issue now. That was resolved because it was a duplicate.
